### ISSUE TITLE
nats ?

### ISSUE THREAD

**Issue by gedw99** [groupcache/discovery-go#1]
Is the intention to put NATS and other "factories" that work with group cache in this repo ?

**Comment by Tochemey** [comment]
@gedw99 Yeah I believe this where we add all the various discovery providers like:

- k8s
- NATS
- etc...

**Comment by thrawn01** [comment]
Yes, exactly! ✨

**Comment by gedw99** [comment]
great stuff. Looking forward to contributing. I am in transit but hope to contribute when I get back Home in a few weeks.

**Comment by Tochemey** [comment]
@thrawn01 Please what is your plan of rolling out the discovery plugin. I have seen you guys did some good job in gubernator. Are you planning to replicate it here to make easy for developers to integrate them. I have seen in gubernator you have implemented at least discovery for:

- k8s
- Memberlist
- DNS
- etcd
